### PR TITLE
plugin/file: Fix print tree error

### PR DESCRIPTION
**Pull request by victory460**
Signed-off-by: xuweiwei <xuweiwei_yewu@cmss.chinamobile.com>

### 1. Why is this pull request needed and what does it do?

The print method designed in the coredns is a perfect debugging tool, but I found a small problem. The steps to reproduce this problem:

https://github.com/coredns/coredns/blob/4c0eacb6574786c8b3db1e5f068ebce96f7545e1/plugin/file/lookup_test.go#L135-L141
Just need to add *zone.Tree.Print()* to L140, like following:

```
func TestLookup(t *testing.T) {
	zone, err := Parse(strings.NewReader(dbMiekNL), testzone, "stdin", 0)
	if err != nil {
		t.Fatalf("Expected no error when reading zone, got %q", err)
	}
	zone.Tree.Print() //add this
	fm := File{Next: test.ErrorHandler(), Zones: Zones{Z: map[string]*Zone{testzone: zone}, Names: []string{testzone}}}
```

Print result before fix,  it just print two row:
```
=== RUN   TestLookup
dname.miek.nl. 
a.miek.nl. www.miek.nl. miek.nl. archive.miek.nl. mx.miek.nl. *.y.miek.nl. ext-cname.miek.nl. srv.miek.nl. *.x.miek.nl. 
```
Print result after fix:
```
=== RUN   TestLookup
dname.miek.nl. 
a.miek.nl. www.miek.nl. 
miek.nl. archive.miek.nl. mx.miek.nl. *.y.miek.nl. 
ext-cname.miek.nl. srv.miek.nl. *.x.miek.nl. 
```
In summary, the print by level of tree can not work correctly,  if the tree has 4 nodes,  it shoud print three row node,  but it print all node in the two row now. 
### 2. Which issues (if any) are related?
No.
### 3. Which documentation changes (if any) need to be made?
No.
### 4. Does this introduce a backward incompatible change or deprecation?
No.

### CODE REVIEW

**Comment by chrisohaver** [comment]
@victory460, you can push new commits to an existing PR without closing and re-opening a new PR for each change.

re: #4945 #4947 #4948 #4955 #4959

**Comment by victory460** [comment]
> @victory460, you can push new commits to an existing PR without closing and re-opening a new PR for each change.
> 
> re: #4945 #4947 #4948 #4955 #4959

OK. Thanks.

**Review by chrisohaver — CHANGES_REQUESTED** [review]
The test provided doesn't have any fail conditions - so unless it crashes, it will always pass.

The unit test should fail if tested against the current code in master, and pass when tested against the fixed code in the PR.

**Comment by victory460** [comment]
> The test provided doesn't have any fail conditions - so unless it crashes, it will always pass.
> 
> The unit test should fail if tested against the current code in master, and pass when tested against the fixed code in the PR.

Thank you for your review:smile: . According to your suggestions, I updated the unit test code.
If you have any other suggestions, I'm willing to improve it. Thanks. @chrisohaver

**Comment by miekg** [comment]
1 more request. Can you put in the git commit what printed wrong now and how that looks with this fix?

**Comment by victory460** [comment]
> 1 more request. Can you put in the git commit what printed wrong now and how that looks with this fix?

Thank you for your review gratefully @miekg . I updated the git commit content and the description of this PR.

**Comment by yongtang** [comment]
@victory460 I noticed that your commit's email is not associated with your GitHub ID. I would suggest you to update the email of the commit, or add the email to GitHub's associated email address as otherwise your work will not be attributed to you.

**Comment by victory460** [comment]
> @victory460 I noticed that your commit's email is not associated with your GitHub ID. I would suggest you to update the email of the commit, or add the email to GitHub's associated email address as otherwise your work will not be attributed to you.

Thanks a lot！According to your suggestion, I added my commit's email address to my github account in email setting now.

**Comment by Rosalieww** [comment]
We also found the bug in this print tree code, can you merge the PR when you have time for it? Thanks.   @miekg @chrisohaver

**Review by chrisohaver — APPROVED** [review]
